src/components/AddCartButton/AddCartButton.component.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  addToCart,
+  calculateTotalCart,
+  calculateTotalProducts,
+} from '../../actions/cart';
+import { useForm } from '../../utils/hooks/useForm/useForm';
+import {
+  AddButton,
+  AddCartContainer,
+  NumberProducts,
+} from './AddCartButton.style';
+const AddCartButton = () => {
+  const dispatch = useDispatch();
+  const { product } = useSelector((state) => state.cart);
+  const {
+    product: { stock },
+  } = useSelector((state) => state.cart);
+  const [{ cart }, handleInputChange] = useForm({
+    cart: 0,
+  });
+  const handleButton = () => {
+    dispatch(addToCart(product.id, product, cart));
+    dispatch(calculateTotalCart());
+    dispatch(calculateTotalProducts());
```

Comment:
This is not a good approach, if there a more places adding products to the cart, we need to repeat these three actions in every place. You should do it in a single dispatch.